kolibri/plugins/coach/assets/src/composables/useResourceSelection.js

```diff
@@ -0,0 +1,155 @@
+import uniqBy from 'lodash/uniqBy';
+import { ref, computed, getCurrentInstance, watch } from 'vue';
+import ContentNodeResource from 'kolibri-common/apiResources/ContentNodeResource';
+import ChannelResource from 'kolibri-common/apiResources/ChannelResource';
+import useFetch from './useFetch';
+
+/**
+ * @typedef {import('../../../../../../composables/useFetch').FetchObject} FetchObject
+ */
+
+/**
+ * Composable for managing the selection of resources within a topic tree.
+ * This utility handles selection rules, manages fetch states for channels, bookmarks,
+ * and topic trees, and offers methods to add, remove, or override selected resources.
+ *
+ * @typedef {Object} UseResourceSelectionResponse
+ * @property {Object} topic Topic tree object, contains the information of the topic,
+ *   its ascendants and children.
+ *   Defined only if the `topicId` query in the route is set.
+ * @property {boolean} loading Indicates whether the main topic tree, channels, and bookmarks
+ *   data are currently loading. This does not account for loading more data. For such cases,
+ *   use the fetch objects of each entity.
+ * @property {FetchObject} channelsFetch Channels fetch object to manage the process of
+ *   fetching channels. We currently don't support fetching more channels.
+ * @property {FetchObject} bookmarksFetch Bookmarks fetch object to manage the process of
+ *   fetching bookmarks. Fetching more bookmarks is supported.
+ * @property {FetchObject} treeFetch Topic tree fetch object to manage the process of
+ *   fetching topic trees and their resources. Fetching more resources is supported.
```

Comment:
Yes, this means pagination. The thing is that this specific endpoint is paginated [in the backend](https://github.com/AlexVelezLl/kolibri/blob/d9b10b615f231011fcb2ec2f24629f1ddb8d8a83/kolibri/core/content/api.py#L965), so its a little more confusing because although we are not passing a "limit", the number of resources that it lists for a specific parentId is paginated, and it comes with a "more" object to fetch the next page.